File: tests/fixtures/vulnerable/prompt_injection_test.js
// aiscan test fixture — should trigger AI-SEC-018 (Prompt Injection, JS)
// DO NOT USE IN PRODUCTION

const Anthropic = require("@anthropic-ai/sdk");
const OpenAI = require("openai");

const anthropic = new Anthropic();
const openai = new OpenAI();

app.post("/chat", async (req, res) => {
  // Vulnerable: template literal interpolation into Anthropic system prompt
  const r1 = await anthropic.messages.create({
    model: "claude-sonnet-4-6",
    max_tokens: 1024,
    system: `You are an assistant. User name: ${req.body.name}`,
    messages: [{ role: "user", content: "hi" }],
  });

  // Vulnerable: string concatenation into system prompt
  const r2 = await anthropic.messages.create({
    model: "claude-sonnet-4-6",
    max_tokens: 1024,
    system: "You are a bot. " + req.query.instructions,
    messages: [{ role: "user", content: "hi" }],
  });

  // Vulnerable: OpenAI system role content built from request body
  const r3 = await openai.chat.completions.create({
    model: "gpt-4o",
    messages: [
      { role: "system", content: `Policy: ${req.body.policy}` },
      { role: "user", content: "hi" },
    ],
  });

  res.json({ r1, r2, r3 });
});

File: tests/fixtures/vulnerable/prompt_injection_test.py
# aiscan test fixture — should trigger AI-SEC-018 (Prompt Injection)
# DO NOT USE IN PRODUCTION

import anthropic
import openai
from flask import request


client = anthropic.Anthropic()

# Vulnerable: f-string interpolation of request data into Anthropic system prompt
resp = client.messages.create(
    model="claude-sonnet-4-6",
    max_tokens=1024,
    system=f"You are a helpful assistant. User name: {request.args['name']}",
    messages=[{"role": "user", "content": "hi"}],
)

# Vulnerable: string concatenation into system prompt
system_prompt = "You are a bot. " + request.json["instructions"]
client.messages.create(
    model="claude-sonnet-4-6",
    max_tokens=1024,
    system=system_prompt,
    messages=[{"role": "user", "content": "hi"}],
)

# Vulnerable: .format() injection into system prompt
client.messages.create(
    model="claude-sonnet-4-6",
    max_tokens=1024,
    system="You are a helpful {} agent".format(request.args.get("persona")),
    messages=[{"role": "user", "content": "hi"}],
)

# Vulnerable: OpenAI messages array with tainted system content
openai_client = openai.OpenAI()
openai_client.chat.completions.create(
    model="gpt-4o",
    messages=[
        {"role": "system", "content": f"System rules: {request.form['policy']}"},
        {"role": "user", "content": "hi"},
    ],
)
